packages/@o3r/extractors/src/utils/config-doc.ts

```diff
@@ -45,6 +48,37 @@ export function getTitleFromDocComment(docComment: DocComment): string | undefin
   return getInlineBlockTagContentFromDocComment(docComment, '@title');
 }
 
+/**
+ * Get widget information from a given DocComment.
+ *
+ *  The widget information are extracted from @widget tag.
+ *
+ * @param docText The tsdoc text to get widget information from
+ */
+export function getWidgetInformationFromDocComment(docText: string): ConfigPropertyWidget | undefined {
+  const widgetInfoText = docText.match(/@o3rWidget (.*)/)?.[1].trim();
+  if (!widgetInfoText) {
+    return;
+  }
+  const firstSpaceIndex = widgetInfoText.indexOf(' ');
+  if (firstSpaceIndex < 1) {
+    return;
+  }
+  const widgetType = widgetInfoText.slice(0, firstSpaceIndex);
+  let widgetParameters: ConfigPropertyWidget['parameters'] = {};
+  const widgetParametersText = widgetInfoText.slice(firstSpaceIndex + 1);
+  try {
+    widgetParameters = JSON.parse(widgetParametersText);
```

Comment:
So we never validate the type of the widget parameter's value?

Reply:
Later in the extraction process there is a schema validation of what is extracted
https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/components/builders/component-extractor/index.ts#L87